src/ezloCap.c

```diff
@@ -0,0 +1,277 @@
+void EzloCap(Entity* this) {
+    if (this->action == 0) {
+        this->action += 1;
+        sub_0805E3A0(this, 2);
+        sub_0807DD50(this);
+    }
+    sub_0807DDAC(this, 0);
+    sub_0807DDE4(this);
+    UpdateAnimationSingleFrame(this);
+}
+
+NONMATCH("asm/non_matching/ezloCap/sub_0806D8A0.inc", void sub_0806D8A0(Entity* this, u32 param_2)) {
+}
+END_NONMATCH
+
+NONMATCH("asm/non_matching/ezloCap/sub_0806D908.inc", void sub_0806D908(Entity* this)) {
+}
+END_NONMATCH
+
+void sub_0806D944(Entity* this) {
+    this->spriteSettings.b.flipX = 0;
+    if (this->x.WORD <= gPlayerEntity.x.WORD) {
+        this->spriteSettings.b.flipX = 1;
+    }
+}
+
+void sub_0806D96C(Entity* this) {
+    Entity* fx = CreateFx(this, 0x42, 0);
+    if (fx != NULL) {
+        fx->spritePriority.b0 = 1;
+        PositionRelative(this, fx, 0, 0xFFE80000);
+    }
+}
+
+void sub_0806D9A4(Entity* this) {
+    if (this->x.WORD <= gPlayerEntity.x.WORD) {
+        gPlayerEntity.spriteSettings.b.flipX = 0;
+    } else {
+        gPlayerEntity.spriteSettings.b.flipX = 1;
+    }
+    gPlayerEntity.animationState = 4;
+}
+
+void NPC4E(Entity* this) {
+    if (this->action == 0) {
+        this->action = 1;
+        this->spriteSettings.b.draw = 4;
+        this->boundingBox = &gUnk_080FD170;
+        sub_0807DD50(this);
+    } else {
+        sub_0807DD94(this, 0);
+    }
+}
+
+NONMATCH("asm/non_matching/ezloCap/sub_0806DA04.inc", void sub_0806DA04(Entity* this, u32 param_2)) {
+    u8 direction =
+        sub_080045D4(this->x.HALF.HI, this->y.HALF.HI, (int)gPlayerEntity.x.HALF.HI, (int)gPlayerEntity.y.HALF.HI);
+    this->direction = direction;
+
+    this->animationState =
+        (this->animationState & 0x80) |
+        gUnk_08114144[(((direction) << 0x18) >> 0x19)]; // somehow the cast from u8 to u32 of direction is an issue here
+}
+END_NONMATCH
+
+void sub_0806DA1C(Entity* this, u32* param_2) {
+    sub_0808091C((ScreenTransitionData*)(gUnk_081141A4)[param_2[1]], (u32)(gUnk_081141E4[param_2[1]]));
+}
+
+// Returns the kingstone id?
+u8 sub_0806DA3C(Entity* this) {
+    u32 result;
+
+    switch (this->entityType.form) {
+        default:
+            result = 0;
+            break;
+        case 1:
+            result = 1;
+            break;
+        case 2:
+            result = 2;
+            break;
+        case 3:
+            result = 3;
+            break;
+        case 4:
+            result = 4;
+            break;
+        case 5:
+            result = 5;
+            break;
+        case 11:
+            result = 9;
+            break;
+        case 6:
+        case 7:
+        case 8:
+        case 9:
+        case 10:
+            result = sub_0801E99C(this);
+            break;
+    }
+    return result;
+}
+
+// Check whether a kingstone fusion is possible and store the result somewhere in param_2?
+void sub_0806DAAC(Entity* this, u32 param_2) {
+    u32 uVar1;
+    uVar1 = CheckKinstoneFused(sub_0806DA3C(this));
+    *(u32*)(param_2 + 0x14) = uVar1;
```

Comment:
Since param_2 gets used as a pointer, it should be that in the signature too
```suggestion
void sub_0806DAAC(Entity* this, u32* param_2) {
    u32 uVar1;
    uVar1 = CheckKinstoneFused(sub_0806DA3C(this));
    param_2[5] = uVar1;
```